infrastructure/src/main/java/org/corfudb/infrastructure/LayoutServer.java

```diff
@@ -198,19 +202,22 @@ public synchronized void handleMessageLayoutPrepare(
         if (!isBootstrapped(msg, ctx, r)) {
             return;
         }
-        Rank prepareRank = new Rank(msg.getPayload().getRank(), msg.getClientID());
-        Rank phase1Rank = getPhase1Rank();
-        Layout proposedLayout = getProposedLayout();
 
-        long serverEpoch = getServerEpoch();
-        if (msg.getPayload().getEpoch() != serverEpoch) {
+        final long msgEpoch = msg.getPayload().getEpoch();
+        final long serverEpoch = getServerEpoch();
+
+        final Rank prepareRank = new Rank(msg.getPayload().getRank(), msg.getClientID());
+        final Rank phase1Rank = getPhase1Rank(msgEpoch);
+
+        if (msgEpoch != serverEpoch) {
```

Comment:
Are these checks really necessary ? 

Its possible that this condition is false, at the time of the check, but becomes true immediately after the evaluation.

Reply:
Its cleaner to do these checks here:

```java
    LAYOUT_PREPARE(12, new TypeToken<CorfuPayloadMsg<LayoutPrepareRequest>>(){}, true),
    LAYOUT_PROPOSE(14, new TypeToken<CorfuPayloadMsg<LayoutProposeRequest>>(){}, true),
```

Reply:
Let me address this whole epoch mess in the next patch. What you are referring to is the message and not LayoutPrepareRequest/LayoutProposeRequest epoch.